### PR TITLE
standardize docs and minor function cleanup

### PR DESCRIPTION
**Pull request by jkvyff**
Ref: #222 
standardize terms, return non-returning reduce, remove unused args

### CODE REVIEW

File: src/arrays/pull.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * Pull removes all of the given values from an array
  *
  * @param {Array} array input array
- * @param {...*} values values to be removed from the array
+ * @param {...*} [values] values to be removed from the array
```
Comment (evanplaice) [inline src/arrays/pull.js]: Rest params can't be optional. Typescript will reject this.

File: src/objects/assign.js [hunk 2]
```diff
@@ -3,7 +3,7 @@
  * already exists, then it is overwritten when merged from left to right.
  *
  * @param {object} object input object
- * @param {...object} sources input object/s
+ * @param {...object} [sources] input source object(s)
```
Comment (evanplaice) [inline src/objects/assign.js]: Rest params can't be optional

File: src/objects/at.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
  * At, creates an array of values corresponding to paths of the object
  *
  * @param {object} object input object
- * @param {...(string|string[])} paths strings describing paths to be returned from an object
+ * @param {...(string|string[])} [paths] strings describing paths to be returned from an object
```
Comment (evanplaice) [inline src/objects/at.js]: Rest params can't be optional

File: src/objects/defaults.js [hunk 4]
```diff
@@ -3,7 +3,7 @@
  * already exists, then the existing one is kept when merged from left to right.
  *
  * @param {object} object input object
- * @param {...object} sources input object/s
+ * @param {...object} [sources] input source object(s)
```
Comment (evanplaice) [inline src/objects/defaults.js]: Rest params...

File: src/objects/defaultsDeep.js [hunk 5]
```diff
@@ -3,7 +3,7 @@
  * being merged recursively and once a property is set, additional values of the same property are ignored.
  *
  * @param {object} object input object
- * @param {...object} sources input object/s
+ * @param {...object} [sources] input source object(s)
```
Comment (evanplaice) [inline src/objects/defaultsDeep.js]: Rest params...

**Comment by jkvyff** [comment]
Getting errors on rebuild about "option 'allowJs' cannot be specified with option 'declaration". Does that need an extra flag?

**Comment by evanplaice** [comment]
No, you need to update Typescript. IIRC, allowing both "allowJS" and "checkJS" was added in 3.6. This is what enabled the ability to generate typings directly from JSDoc.

#BleedingEdgeProblems

**Comment by jkvyff** [comment]
Ah, I found it was mentioned a couple months ago. Should be good now.

**Comment by evanplaice** [comment]
What's `identity` here?

```javascript
/*  * @param {function} [predicate = identity] optional function to test against object values */
```

Can we just leave the default value out altogether?

**Comment by evanplaice** [comment]
IMO, `undefined` can be assumed to be the default. It doesn't need to appear in the docs.

```javascript
/*  * @param {*} [thisArg = undefined] optional this in a function call */
```

**Comment by jkvyff** [comment]
Good catch, identity to me just meant returns each value of self when iterated, but that didn't become a practice for the library. In the places it could be used, I opted to just return the original object if nothing was happening to the object as it would be much faster. 

For the second one, thisArg being undefined is more of an implementation detail and probably wouldn't help a user understand the function any more.

Removed both from documentation, removed redundant set as undefined for thisArg in implementation.

**Comment by evanplaice** [comment]
Is this ready to land?

**Comment by jkvyff** [comment]
Yep yep

**Pull request merged**